app/src/main/java/com/km/real_convenience_store/ui/product_search/ProductSearchActivity.kt

```diff
@@ -22,33 +23,38 @@ import kotlinx.coroutines.withContext
 class ProductSearchActivity : AppCompatActivity() {
     private lateinit var binding: ActivityProductSearchBinding
     private val productSearchAdapter = ProductSearchAdapter()
-
-    private var saleType: String? = null
-    private var currentPage: Int = 1
-    private var needLoadMore: Boolean = true
+    private val viewModel: ProductSearchViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityProductSearchBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        initViews()
 
+        bindViews()
+        bindObserverData()
         getFavoriteProduct()
     }
 
+    private fun bindObserverData() {
+       viewModel.searchProducts.observe(this){ products ->
+           productSearchAdapter.addProducts(products)
+           binding.tvSearchResultCount.text =
+               "검색결과 ".plus("${productSearchAdapter.itemCount}").plus("건")
```

Comment:
요거 왜 plus 쓴거야?? 
`"검색결과 ${productSearchAdapter.itemCount}건"`
이렇게 쓰는거랑 특별한 차이가 있나??

Reply:
사실 이 부분은 따로 수정은 하지 않았어 기존에 개발되었던대로 사용했던거고 리뷰보고 확인해본결과 

"검색결과 ${productSearchAdapter.itemCount}건"
"검색결과" + productSearchAdapter.itemCount + "건" 
 "검색결과 ".plus("${productSearchAdapter.itemCount}").plus("건")

3개중 뭐를 써도 전부 동일하게 텍스트뷰에 표출되는데 
위의 두개는 textview 직접적인 하드 코딩을 해서 텍스트를  넣어주는거라 린트가 표시가 되더라구
아래에 껄로하면 자바에서 concat 과 같은 기능으로 string을 더해서 텍스트뷰에 보여주는 방식이더라고 이 차이인거 같아

Reply:
저도 그래서 plus로 했었는데 변수로 따로 뽑아서 문자열을 만들어준 다음에 하면 warnning이 안나오더라구요
참고용으로 알아두셔요~

```suggestion
           val result = "검색결과 ${productSearchAdapter.itemCount}건"
           binding.tvSearchResultCount.text = result
```